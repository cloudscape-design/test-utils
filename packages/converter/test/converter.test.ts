// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import fs from 'fs';
import path from 'path';
import convertToSelectorUtil from '../src/index';

const inputDir = path.join(__dirname, 'inputs', 'converter');

fs.readdirSync(inputDir).forEach(name => {
  test(path.basename(name, '.ts'), () => {
    const source = fs.readFileSync(path.join(inputDir, name), 'utf-8').trim();
    const result = convertToSelectorUtil(source);
    expect(result).toMatchSnapshot();
  });
});
