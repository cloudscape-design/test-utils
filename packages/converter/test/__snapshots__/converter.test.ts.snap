// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`multi 1`] = `
"// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { ComponentWrapper, ElementWrapper } from \\"@cloudscape-design/test-utils-core/selectors\\";
import ChildWrapper from './simple';
export default class DummyWrapper extends ComponentWrapper {
  findElement() {
    return this.find('.awsui-element');
  }
  findElements() {
    return this.findAll('.awsui-element');
  }
  findChildren() {
    return this.findAll('.awsui-element').map(element => new ChildWrapper(element.getElement()));
  }
}"
`;

exports[`simple 1`] = `
"// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { ComponentWrapper, ElementWrapper } from \\"@cloudscape-design/test-utils-core/selectors\\";
export default class DummyWrapper extends ComponentWrapper {
  findElement() {
    return this.find('.awsui-element')!;
  }
}"
`;

exports[`strip-dom 1`] = `
"// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
/* eslint-env browser */
import { Simulate } from 'react-dom/test-utils';
import { ComponentWrapper, ElementWrapper } from \\"@cloudscape-design/test-utils-core/selectors\\";
export default class DummyWrapper extends ComponentWrapper {
  findOptionalElement() {
    return this.find('.awsui-optional-element');
  }
  findRequiredElement() {
    return this.find('.awsui-required-element')!;
  }
}"
`;

exports[`strip-imports 1`] = `
"// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0
import { ComponentWrapper } from \\"@cloudscape-design/test-utils-core/selectors\\";
import { KeyCode } from \\"@cloudscape-design/test-utils-core/utils\\";
import ChildWrapper from './simple';
export default class DummyWrapper extends ComponentWrapper {
  findElement() {
    return new ChildWrapper(this.find('.awsui-child')!.getElement());
  }
  findSomething() {
    return KeyCode.enter;
  }
}"
`;
